### DISCUSSION TITLE
About "functional.reset_net(net) "

### DISCUSSION THREAD

**Discussion by mountains-high** [fangwei123456/SpikingFlow#175] [Q&A] [answered]
Hello! Dear all,

I could not find the mathematical model explaining why we use “reset_net(net).” Could you reference some source materials or papers about this functionality, please?

Thank you very much for your consideration and time. 
Have a nice day!

**Comment by fangwei123456** [comment, marked as answer]
·reset_net(net)· will call each module's `reset` function in `net` as long as this module has a `reset` function. All stateful modules in SpikingJelly is the inherited from `clock_driven.base.MemoryModule`. `reset(m)` will reset the hidden states of the module`m` to the default value. For example, if you reset a neuron, then the neuron's voltage will be set to v_reset (if you use hard reset, else it will be set to 0).

You can read the codes of 'base' in SpikingJelly to understand how reset works: https://github.com/fangwei123456/spikingjelly/blob/master/spikingjelly/clock_driven/base.py .
